include/sparrow/array_data_concepts.hpp

```diff
@@ -0,0 +1,41 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "sparrow/dictionary_encoded_layout.hpp"
+#include "sparrow/fixed_size_layout.hpp"
+#include "sparrow/mp_utils.hpp"
+#include "sparrow/variable_size_binary_layout.hpp"
+
+namespace sparrow
+{
+    template <class Layout>
```

Comment:
For each concept, add a comment clarifying what is the intent. That will help with improving these concepts later. See `mp_utils.hpp` for some examples.

Reply:
Done